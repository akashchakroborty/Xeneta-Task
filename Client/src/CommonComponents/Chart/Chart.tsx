import React from 'react';
import {
 LineChart, Line, CartesianGrid, XAxis, YAxis, Tooltip
} from 'recharts';
import { Rates } from '../../Redux/Types/portBasedRatesTypes';

export interface ChartProps {
  data: Rates[];
  dataKeyY: string;
  dataKeyX: string;
  width: number;
  height: number;
  type?:
    | 'step'
    | 'basis'
    | 'basisClosed'
    | 'basisOpen'
    | 'linear'
    | 'linearClosed'
    | 'natural'
    | 'monotoneX'
    | 'monotoneY'
    | 'monotone'
    | 'stepBefore'
    | 'stepAfter'
    | undefined;
}

const ChartComponent: React.FC<ChartProps> = ({
  data,
  dataKeyY,
  dataKeyX,
  width,
  height,
  type = 'monotone',
}) => {
  return (
    <LineChart
      width={width}
      height={height}
      data={data}
      margin={{
 top: 5, right: 20, bottom: 5, left: 0
}}
    >
      <Line type={type} dataKey={dataKeyY} stroke="#00539f" />
      <CartesianGrid stroke="#00539f" strokeDasharray="5 5" vertical={false} />
      <XAxis dataKey={dataKeyX} />
      <YAxis domain={['dataMin', 'dataMax']} />
      <Tooltip />
    </LineChart>
  );
};

export default ChartComponent;
