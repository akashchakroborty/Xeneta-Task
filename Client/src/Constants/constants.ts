export const SELECT_ORIGIN = 'Select Origin';
export const SELECT_DESTINATION = 'Select Destination';
export const EMPTY_STRING = '';
export const DEFAULT_PORTS = [];
export const DEFAULT_RATES = [];
export const DEFAULT_LOADING = false;
