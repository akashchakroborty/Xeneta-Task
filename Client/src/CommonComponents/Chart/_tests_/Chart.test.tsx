import React from 'react';
import { shallow } from 'enzyme';
import Chart, { ChartProps } from '../Chart';
import { MOCK_RATES_DATA } from '../../../Constants/testData';

describe('should test the Chart Component', () => {
  const mockProps: ChartProps = {
    data: MOCK_RATES_DATA,
    dataKeyY: 'low',
    dataKeyX: 'day',
    width: 1300,
    height: 400,
  };
  test('should render the Component', () => {
    const component = shallow(<Chart {...mockProps} />);
    expect(component).toMatchSnapshot();
  });
});
