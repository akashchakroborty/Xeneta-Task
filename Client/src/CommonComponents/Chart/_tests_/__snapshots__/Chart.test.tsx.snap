// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the Chart Component should render the Component 1`] = `
<LineChart
  barCategoryGap="10%"
  barGap={4}
  data={
    Array [
      Object {
        "day": "2021-01-01",
        "high": 2436,
        "low": 1037,
        "mean": 1615,
      },
      Object {
        "day": "2021-01-02",
        "high": 2436,
        "low": 1037,
        "mean": 1615,
      },
      Object {
        "day": "2021-01-03",
        "high": 2436,
        "low": 1037,
        "mean": 1615,
      },
      Object {
        "day": "2021-01-04",
        "high": 2435,
        "low": 1037,
        "mean": 1615,
      },
      Object {
        "day": "2021-01-05",
        "high": 2427,
        "low": 1037,
        "mean": 1615,
      },
      Object {
        "day": "2021-01-06",
        "high": 2431,
        "low": 1037,
        "mean": 1615,
      },
      Object {
        "day": "2021-01-07",
        "high": 2437,
        "low": 1037,
        "mean": 1615,
      },
      Object {
        "day": "2021-01-08",
        "high": 2438,
        "low": 1036,
        "mean": 1619,
      },
      Object {
        "day": "2021-01-09",
        "high": 2438,
        "low": 1036,
        "mean": 1619,
      },
      Object {
        "day": "2021-01-10",
        "high": 2438,
        "low": 1037,
        "mean": 1615,
      },
      Object {
        "day": "2021-01-11",
        "high": 2436,
        "low": 1037,
        "mean": 1615,
      },
      Object {
        "day": "2021-01-12",
        "high": 2436,
        "low": 1036,
        "mean": 1614,
      },
      Object {
        "day": "2021-01-13",
        "high": 2436,
        "low": 1036,
        "mean": 1614,
      },
      Object {
        "day": "2021-01-14",
        "high": 2436,
        "low": 1036,
        "mean": 1614,
      },
      Object {
        "day": "2021-01-15",
        "high": 2438,
        "low": 1036,
        "mean": 1614,
      },
      Object {
        "day": "2021-01-16",
        "high": 2437,
        "low": 1037,
        "mean": 1611,
      },
      Object {
        "day": "2021-01-17",
        "high": 2437,
        "low": 1037,
        "mean": 1611,
      },
      Object {
        "day": "2021-01-18",
        "high": 2437,
        "low": 1037,
        "mean": 1611,
      },
      Object {
        "day": "2021-01-19",
        "high": 2437,
        "low": 1037,
        "mean": 1611,
      },
      Object {
        "day": "2021-01-20",
        "high": 2437,
        "low": 1037,
        "mean": 1611,
      },
      Object {
        "day": "2021-01-21",
        "high": 2436,
        "low": 1037,
        "mean": 1611,
      },
      Object {
        "day": "2021-01-22",
        "high": 2434,
        "low": 1037,
        "mean": 1611,
      },
      Object {
        "day": "2021-01-23",
        "high": 2434,
        "low": 1037,
        "mean": 1611,
      },
      Object {
        "day": "2021-01-24",
        "high": 2434,
        "low": 1037,
        "mean": 1611,
      },
      Object {
        "day": "2021-01-25",
        "high": 2435,
        "low": 1037,
        "mean": 1611,
      },
      Object {
        "day": "2021-01-26",
        "high": 2436,
        "low": 1037,
        "mean": 1611,
      },
      Object {
        "day": "2021-01-27",
        "high": 2437,
        "low": 1037,
        "mean": 1611,
      },
      Object {
        "day": "2021-01-28",
        "high": 2438,
        "low": 1037,
        "mean": 1611,
      },
      Object {
        "day": "2021-01-29",
        "high": 2435,
        "low": 1037,
        "mean": 1611,
      },
      Object {
        "day": "2021-01-30",
        "high": 2435,
        "low": 1037,
        "mean": 1611,
      },
      Object {
        "day": "2021-01-31",
        "high": 2435,
        "low": 1037,
        "mean": 1611,
      },
    ]
  }
  height={400}
  layout="horizontal"
  margin={
    Object {
      "bottom": 5,
      "left": 0,
      "right": 20,
      "top": 5,
    }
  }
  reverseStackOrder={false}
  stackOffset="none"
  width={1300}
>
  <Line
    activeDot={true}
    animateNewValues={true}
    animationBegin={0}
    animationDuration={1500}
    animationEasing="ease"
    connectNulls={false}
    dataKey="low"
    dot={true}
    fill="#fff"
    hide={false}
    isAnimationActive={true}
    legendType="line"
    points={Array []}
    stroke="#00539f"
    strokeWidth={1}
    type="monotone"
    xAxisId={0}
    yAxisId={0}
  />
  <CartesianGrid
    fill="none"
    horizontal={true}
    horizontalFill={Array []}
    horizontalPoints={Array []}
    stroke="#00539f"
    strokeDasharray="5 5"
    vertical={false}
    verticalFill={Array []}
    verticalPoints={Array []}
  />
  <XAxis
    allowDataOverflow={false}
    allowDecimals={true}
    allowDuplicatedCategory={true}
    dataKey="day"
    domain={
      Array [
        0,
        "auto",
      ]
    }
    height={30}
    hide={false}
    mirror={false}
    orientation="bottom"
    padding={
      Object {
        "left": 0,
        "right": 0,
      }
    }
    reversed={false}
    scale="auto"
    tickCount={5}
    type="category"
    width={0}
    xAxisId={0}
  />
  <YAxis
    allowDataOverflow={false}
    allowDecimals={true}
    allowDuplicatedCategory={true}
    domain={
      Array [
        "dataMin",
        "dataMax",
      ]
    }
    height={0}
    hide={false}
    mirror={false}
    orientation="left"
    padding={
      Object {
        "bottom": 0,
        "top": 0,
      }
    }
    reversed={false}
    scale="auto"
    tickCount={5}
    type="number"
    width={60}
    yAxisId={0}
  />
  <Tooltip
    active={false}
    allowEscapeViewBox={
      Object {
        "x": false,
        "y": false,
      }
    }
    animationDuration={400}
    animationEasing="ease"
    contentStyle={Object {}}
    coordinate={
      Object {
        "x": 0,
        "y": 0,
      }
    }
    cursor={true}
    cursorStyle={Object {}}
    filterNull={true}
    isAnimationActive={true}
    itemStyle={Object {}}
    labelStyle={Object {}}
    offset={10}
    separator=" : "
    trigger="hover"
    useTranslate3d={false}
    viewBox={
      Object {
        "x1": 0,
        "x2": 0,
        "y1": 0,
        "y2": 0,
      }
    }
    wrapperStyle={Object {}}
  />
</LineChart>
`;
